pyxem/tests/signals/test_diffraction2d.py

```diff
@@ -491,6 +491,160 @@ def test_lazy_input_lazy_result(self):
         s_shift.compute()
 
 
+class TestCenterDirectBeam:
+    def setup_method(self):
+        data = np.zeros((8, 6, 20, 16), dtype=np.int16)
+        x_pos_list = np.random.randint(8 - 2, 8 + 2, 6, dtype=np.int16)
+        x_pos_list[x_pos_list == 8] = 9
+        y_pos_list = np.random.randint(10 - 2, 10 + 2, 8, dtype=np.int16)
+        for ix in range(len(x_pos_list)):
+            for iy in range(len(y_pos_list)):
+                data[iy, ix, y_pos_list[iy], x_pos_list[ix]] = 9
+        x_pos_list = x_pos_list
+        y_pos_list = y_pos_list
+        s = Diffraction2D(data)
+        s.axes_manager[0].scale = 0.5
+        s.axes_manager[1].scale = 0.6
+        s.axes_manager[2].scale = 3
+        s.axes_manager[3].scale = 4
+        s_lazy = s.as_lazy()
+        self.s = s
+        self.s_lazy = s_lazy
+        self.x_pos_list = x_pos_list
+        self.y_pos_list = y_pos_list
+
+    def test_non_lazy(self):
+        s = self.s
+        s.center_direct_beam(method="blur", sigma=1)
+        assert s._lazy is False
+        assert (s.data[:, :, 10, 8] == 9).all()
+        s.data[:, :, 10, 8] = 0
+        assert not s.data.any()
```

Comment:
What does this do?

Reply:
Basically making sure none of the other pixel values have changed. I added a comment the first time this occurs, so anyone reading the code in the future will understand it.

Reply:
oh it's two lines of stuff, I see, you set the spot to 0 and then assert that all the pixels are close to zero (ie-unchanged).